packages/core/src/clients/ILeaderboardClient.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  LeaderboardDelete,
+  LeaderboardFetch,
+  LeaderboardGetRank,
+  LeaderboardLength,
+  LeaderboardRemoveElements,
+  LeaderboardUpsert,
+} from '../messages/responses/leaderboard';
+import {
+  SortedSetFetchByRankCallOptions,
+  SortedSetFetchByScoreCallOptions,
+} from '../utils';
+
+export type LeaderboardFetchByRankOptions = SortedSetFetchByRankCallOptions;
+export type LeaderboardFetchByScoreOptions = SortedSetFetchByScoreCallOptions;
+
+export interface ILeaderboardClient {
+  leaderboardUpsert(
+    cacheName: string,
+    leaderboardName: string,
+    elements: Map<bigint, number>
```

Comment:
I wonder if we should consider making this `Map<bigint | number, number>`? On the output side we have to do `bigint` or we risk losing precision, but on the input side we can potentially make it more flexible for people?